smac/intensification/successive_halving.py

```diff
@@ -820,3 +872,70 @@ def _top_k(self,
         # select top configurations only
         top_configs = configs_sorted[:k]
         return top_configs
+
+    def _count_running_instances_for_challenger(self, run_history: RunHistory) -> int:
+        """
+        The intensifiers are called on a sequential manner. In each iteration,
+        one can only return a configuration at a time, for that reason
+        self.running_challenger tracks that more instance/seed pairs need to be
+        launched for a given config.
+
+        This procedure counts the number of running instances/seed pairs for the
+        current running challenger
+        """
+        running_instances = 0
+
+        if self.running_challenger is not None:
+            for k, v in run_history.data.items():
+                if run_history.ids_config[k.config_id] == self.running_challenger:
+                    if v.status == StatusType.RUNNING:
+                        running_instances += 1
+
+        return running_instances
+
+    def _launched_all_configs_for_current_stage(self, run_history: RunHistory) -> bool:
+        """
+        This procedure queries if the addition of currently finished configs
+        and running configs are sufficient for the current stage.
+        If more configs are needed, it will return False.
+        Parameters
+        ----------
+        run_history : RunHistory
+            stores all runs we ran so far
+
+        Returns
+        -------
+            bool: Whether or not to launch more configurations/instances/seed pairs
+        """
+        # selecting instance-seed subset for this budget, depending on the kind of budget
+        curr_budget = self.all_budgets[self.stage]
+        if self.instance_as_budget:
+            prev_budget = int(self.all_budgets[self.stage - 1]) if self.stage > 0 else 0
+            curr_insts = self.inst_seed_pairs[int(prev_budget):int(curr_budget)]
+        else:
+            curr_insts = self.inst_seed_pairs
+
+        # _count_running_instances_for_challenger will count the running instances
+        # of the last challenger. It makes sense here, because we assume that if we
+        # moved to a new challenger, all instances have been launched for a previous
+        # challenger
+        running_instances = self._count_running_instances_for_challenger(run_history)
+        n_insts_remaining = len(curr_insts) - (self.curr_inst_idx + running_instances)
+
+        # Check which of the current configs is running
+        my_configs = [c for c, i, s in self.run_tracker]
+        running_configs = set()
+        for k, v in run_history.data.items():
+            if v.status == StatusType.RUNNING:
+                if run_history.ids_config[k.config_id] in my_configs:
+                    running_configs.add(k.config_id)
+
+        # if all configurations for the current stage have been evaluated, reset stage
```

Comment:
There's no reset happening here, right? Could you please drop this from the comment?

Reply:
Sorry, copy paste issue that is now fixed